server/src/Q3Server/AuthParameterChecker.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Owin;
+
+namespace Q3Server
+{
+    public class AuthParameterChecker : OwinMiddleware
+    {
+        public AuthParameterChecker(OwinMiddleware next) : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            if (HasCorrectParameterCombination(context.Request))
+            {
+                await Next.Invoke(context);
+            }
+            else
+            {
+                context.Response.StatusCode = 400;

```

Comment:
That article reveals an interesting OWIN middleware gotcha - "once you start writing to the response body, you can no longer write to the response headers"!